hydroflow_lang/src/graph/flat_graph_builder.rs

```diff
@@ -888,4 +889,42 @@ impl FlatGraphBuilder {
             }
         }
     }
+
+    /// Operators with singleton references must be preceeded by a `persist()` in their pipeline,
+    /// to enable replay when the singleton updates.
+    fn ensure_singleton_referencers_succeed_persist(&mut self) {
+        // Resolve the singleton references for each node.
+        for node_id in self.flat_graph.node_ids().collect::<Vec<_>>() {
+            let GraphNode::Operator(operator) = self.flat_graph.node(node_id) else {
+                continue;
```

Comment:
Curious about what you feel about using `.filter()` on iterators.`

Reply:
Not against it, in this case it's easier with the for loop since it grabs the `operator` and `operator.span()`